src/dcm230/dcm230.py

```diff
@@ -0,0 +1,243 @@
+
+    For each entry in `cls._register_specs`, this decorator dynamically creates
+    a corresponding @property getter, and optionally a setter if `writable=True`.
+
+    The generated getter automatically calls `_read_register(register_name)`
+    and the setter calls `_write_register(address, value)` with range validation
+    if enabled in the `RegisterSpec`.
+
+    Args:
+        cls: The target class to which properties will be added.
+
+    Returns:
+        The same class with dynamically added properties.
+    """
+    for name, spec in cls._register_specs.items():
+
+        def getter(
+            self: "Dcm230", _name: str = name, _spec: "RegisterSpec" = spec
+        ) -> Decimal | int:
+            """Auto-generated register reader.
+
+            Returns the current value of the register. If range validation is
+            enabled, ensures that the returned value is within the expected range.
+
+            Raises:
+                ValueError: If the register value is outside its defined range.
+            """
+            _value = self._read_register(_name)
+            if _spec.range and not (_spec.min <= _value <= _spec.max):
+                msg = f"Invalid value for '{_name}': {_value}. Must be between {_spec.min} and {_spec.max}."
+                raise ValueError(msg)
+            return _value
+
+        def setter(
+            self: "Dcm230", value: int, _name: str = name, _spec: "RegisterSpec" = spec
+        ) -> None:
+            """Auto-generated register writer.
+
+            Writes a new value to the register and performs range validation
+            if defined in the corresponding `RegisterSpec`.
+
+            Raises:
+                AttributeError: If the register is read-only.
+                ValueError: If the written value is outside its defined range.
+            """
+            if not _spec.writable:
+                msg = f"Register '{_name}' is read-only."
+                raise AttributeError(msg)
+
+            if _spec.range and not (_spec.min <= value <= _spec.max):
+                msg = f"Invalid value for '{_name}': {value}. Must be between {_spec.min} and {_spec.max}."
+                raise ValueError(msg)
+            self._write_register(_spec.address, int(value))
+
+        prop = property(getter, setter) if spec.writable else property(getter)
+
+        prop.__doc__ = f"{name} ({'read/write' if spec.writable else 'read-only'})" + (
+            f" range=[{spec.min}, {spec.max}]" if spec.range else ""
+        )
+
+        setattr(cls, name, prop)
+
+    return cls
+
+
+@register_properties
+class Dcm230:
+    """Driver for Eastron Dcm230 series energy meters.
+
+    Provides read and write access to Modbus registers via an existing
+    `pymodbus.client.ModbusSerialClient` instance. Register definitions are
+    dynamically mapped to @property accessors based on `_register_specs`.
+    """
+
+    INT16_REG_COUNT = 1
+    INT32_REG_COUNT = 2
```

Comment:
When looking at the register list for DCM230, it only appears to have one 16-bit register? Could be worth to implement the outlier and simplify the golden path by removing the 16bit checks.